### PR TITLE
Client10

### PR DESCRIPTION
**Pull request by ambid17**
Added item/character controlling in the dev console, i combined ticket 10/11 so i could test making it generic, so it works for future tables easier

### CODE REVIEW

File: DungeonMaster/Assets/Scripts/DeveloperConsole.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        StartCoroutine(LOADSTUFF(baseURL + currentDataType));
+    }
+
+    void CleanScrollView() {
+        foreach(Transform child in scrollViewContent.transform) {
+            Destroy(child.gameObject);
+        }
+    }
+
+    IEnumerator LOADSTUFF(string url) {
```
Comment (0xLeif) [inline DungeonMaster/Assets/Scripts/DeveloperConsole.cs]: NAHSTUFF

File: DungeonMaster/Assets/Scripts/DeveloperConsole.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        string json = "";
+
+        switch (currentDataType) {
+            case DataType.items:
+                Item item = new Item(null, "s", "s", "s", "s", 1, new Currency(0, 0, 0, 0, 0), 0);
+                json = JsonUtility.ToJson(item);
+                break;
+            case DataType.characters:
+                Stats stats = new Stats(0, 0, 0, 0, 0, 0);
+                Skills skills = new Skills(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
```
Comment (0xLeif) [inline DungeonMaster/Assets/Scripts/DeveloperConsole.cs]: So for these that have default values make a static var for the class like Skills.default

File: DungeonMaster/Assets/Scripts/DeveloperConsole.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                Character character = new Character(null, "leif", 0, "Dwarf", "Mine", 0, 0, 10, 0, 0, 0, 0, new Ability[] { }, new string[] { }, new string[] { }, new string[] { }, new string[] { }, skills, stats, stats);
+                json = JsonUtility.ToJson(character);
+                break;
+        }
+
+        //TODO figure out how to remove the ID from JSON
```
Comment (0xLeif) [inline DungeonMaster/Assets/Scripts/DeveloperConsole.cs]: Todo

**Review by 0xLeif — CHANGES_REQUESTED** [review]

File: DungeonMaster/Assets/Scripts/DeveloperConsole.cs (hunk 2 above)
Comment (ambid17) [reply DungeonMaster/Assets/Scripts/DeveloperConsole.cs]: oooh i was wondering how to do this cleanly

File: DungeonMaster/Assets/Scripts/HomeUIManager.cs [hunk 4]
```diff
@@ -19,10 +19,12 @@ public class HomeUIManager : MonoBehaviour
     public GameObject tutorialPanel;
 
     public GameObject mainPanel;
+
+    public GameObject adminPanel;
     #endregion
 
     //TODO: Extrapolate these out to a constants file or URL builder
-    const string baseURL = "dungeonmaster-development.vapor.cloud";
+    public string baseURL = "dungeonmaster-development.vapor.cloud";
```
Comment (parshav) [inline DungeonMaster/Assets/Scripts/HomeUIManager.cs]: Could this technically still be a const public value ?

File: DungeonMaster/Assets/Scripts/HomeUIManager.cs [hunk 5]
```diff
@@ -112,18 +129,18 @@ void Start()
 
         yield return StartCoroutine(WaitForRequest(www, loginRoute));
         GoToMain();
-        StartCoroutine(PROFILE());
+        StartCoroutine(ProfileRequest());
     }
 
-    IEnumerator PROFILE() {
+    IEnumerator ProfileRequest() {
         UnityWebRequest www = UnityWebRequest.Get(URLPrefix + baseURL + "/" + profileRoute);
```
Comment (parshav) [inline DungeonMaster/Assets/Scripts/HomeUIManager.cs:136]: Not a change on this pr but would suggest a function for building the url for these, as there will be more endpoints coming up.

File: DungeonMaster/Assets/Scripts/NetworkingCalls.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            GlobalDebug.LogMessage("request failed, url: " + www.url + " error: " + www.error + " body: " + System.Text.Encoding.UTF8.GetString(www.uploadHandler.data));
+        } else {
+            GlobalDebug.LogMessage("request succeeded, url: " + www.url + " responseCode: " + www.responseCode + " body: " + www.downloadHandler.text);
+        }
+        JsonUtility.FromJson<Type>(www.downloadHandler.text);
+    }
+    
+    string GetJSONWithoutID(object o) {
+        string json = JsonUtility.ToJson(o);
+        return json.Replace("\"\"", "null");
```
Comment (parshav) [inline DungeonMaster/Assets/Scripts/NetworkingCalls.cs]: Does this technically still pass a string with the value `null` in the field your replacing? Was that the intent?

File: DungeonMaster/Assets/Scripts/Models/Skills.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+[Serializable]
+public class Skills
+{
+    public static Skills defaultSkills = new Skills(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
```
Comment (parshav) [inline DungeonMaster/Assets/Scripts/Models/Skills.cs:9]: Would highly suggest implementing the builder pattern for complex constructors, eg  http://www.stefanoricciardi.com/2010/04/14/a-fluent-builder-in-c/

**Review by parshav — CHANGES_REQUESTED** [review]

File: DungeonMaster/Assets/Scripts/HomeUIManager.cs (hunk 5 above)
Comment (ambid17) [reply DungeonMaster/Assets/Scripts/HomeUIManager.cs:136]: im working on figuring out a static class to handle all of the networking

File: DungeonMaster/Assets/Scripts/HomeUIManager.cs (hunk 4 above)
Comment (ambid17) [reply DungeonMaster/Assets/Scripts/HomeUIManager.cs]: you right

File: DungeonMaster/Assets/Scripts/HomeUIManager.cs (hunk 5 above)
Comment (0xLeif) [reply DungeonMaster/Assets/Scripts/HomeUIManager.cs:136]: https://github.com/neuecc/UniRx

Comment (parshav) [reply DungeonMaster/Assets/Scripts/HomeUIManager.cs:136]: gg.png ?

Comment (ambid17) [reply DungeonMaster/Assets/Scripts/HomeUIManager.cs:136]: https://github.com/oneleif/DungeonMaster-Client/issues/15

File: DungeonMaster/Assets/Scripts/Models/Skills.cs (hunk 7 above)
Comment (ambid17) [reply DungeonMaster/Assets/Scripts/Models/Skills.cs:9]: https://github.com/oneleif/DungeonMaster-Client/issues/14

File: DungeonMaster/Assets/Scripts/DeveloperConsole.cs (hunk 3 above)
Comment (ambid17) [reply DungeonMaster/Assets/Scripts/DeveloperConsole.cs]: https://github.com/oneleif/DungeonMaster-Client/issues/16

File: DungeonMaster/Assets/Scripts/NetworkingCalls.cs (hunk 6 above)
Comment (ambid17) [reply DungeonMaster/Assets/Scripts/NetworkingCalls.cs]: https://github.com/oneleif/DungeonMaster-Client/issues/16

**Review by 0xLeif — APPROVED** [review]

**Review by parshav — APPROVED** [review]

**Pull request merged**